src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,17 +1,36 @@
 import React from 'react';
+import { SortType } from '../../types/SortType';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { actions as filterActions } from '../../features/filter';
 
 export const TodoFilter: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const query = useAppSelector(state => state.filter.query);
+  const sortType = useAppSelector(state => state.filter.status);
+
+  const hadnleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    dispatch(filterActions.setFilter(query, event.target.value as SortType));
+  };
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    dispatch(filterActions.setFilter(event.target.value, sortType));
+  };
+
   return (
     <form
       className="field has-addons"
       onSubmit={event => event.preventDefault()}
     >
       <p className="control">
         <span className="select">
-          <select data-cy="statusSelect">
-            <option value="all">All</option>
-            <option value="active">Active</option>
-            <option value="completed">Completed</option>
+          <select
+            data-cy="statusSelect"
+            onChange={hadnleSelectChange}
+            value={sortType}
+          >
+            <option value={SortType.all}>All</option>
```

Comment:
```suggestion
            <option value={SortType.All}>all</option>
```
PascalCase is ok for enum keys

The requested change was about text you are displaying :)